web/angular-wallet/src/app/main/transactions/transaction-details/transaction-details.component.ts

```diff
@@ -1,121 +1,76 @@
-import {Component, OnInit, ChangeDetectionStrategy} from '@angular/core';
-import {MatDialogRef} from '@angular/material/dialog';
+import {Component, OnInit} from '@angular/core';
 import {
-  EncryptedMessage,
-  Message,
   Account,
-  Transaction,
-  TransactionType,
-  TransactionSmartContractSubtype, assertAttachmentVersion
+  Transaction
 } from '@burstjs/core';
 import {StoreService} from 'app/store/store.service';
 import {ActivatedRoute} from '@angular/router';
 import {UtilService} from '../../../util.service';
 import {AccountService} from 'app/setup/account/account.service';
-import {convertHexStringToString} from '@burstjs/util';
-
-type TransactionDetailsCellValue = string | Message | EncryptedMessage | number;
-type TransactionDetailsCellValueMap = [string, TransactionDetailsCellValue];
+import {CellValue, CellValueMapper} from './cell-value-mapper';
+import {I18nService} from '../../../layout/components/i18n/i18n.service';
+import {takeUntil} from 'rxjs/operators';
+import {UnsubscribeOnDestroy} from '../../../util/UnsubscribeOnDestroy';
+
+export interface TransactionDetailRow {
+  k: string;
+  l: string;
+  v: CellValue;
+}
 
 @Component({
   selector: 'app-transaction-details',
   templateUrl: './transaction-details.component.html',
-  styleUrls: ['./transaction-details.component.css']
+  styleUrls: ['./transaction-details.component.scss']
 })
 export class TransactionDetailsComponent implements OnInit {
 
-  infoColumns = ['key', 'value'];
   infoRows = ['transactionType', 'attachment', 'amountNQT', 'feeNQT', 'senderAddress', 'recipientAddress'];
-  infoData: Map<string, TransactionDetailsCellValue>;
-  detailsData: Map<string, TransactionDetailsCellValue>;
+  public detailsData: TransactionDetailRow[] = [];
   account: Account;
   transaction: Transaction;
   recipient: Account;
+  private cellValueMapper: CellValueMapper;
 
   constructor(
     private storeService: StoreService,
     private utilService: UtilService,
     private route: ActivatedRoute,
-    private accountService: AccountService) {
-  }
-
-  private getNameFromTransactionSubtype(): string {
-    return this.utilService.translateTransactionSubtype(this.transaction, this.account);
-  }
-
-  private getNameFromTransactionType(): string {
-    return this.utilService.translateTransactionType(this.transaction);
-  }
-
-  closeDialog(): void {
-  }
-
-  public getInfoData(): TransactionDetailsCellValueMap[] {
-    return Array.from(this.infoData.entries());
-  }
+    private accountService: AccountService,
+  ) {}
 
-  public getDetailsData(): TransactionDetailsCellValueMap[] {
-    return Array.from(this.detailsData.entries());
-  }
+  async ngOnInit(): Promise<void> {
```

Comment:
I have still problems with translations, i.e. if changing the language it will be updated on reload only